gogol/src/Network/Google/Internal/Auth.hs

```diff
@@ -127,7 +127,14 @@ data AuthorizedUser = AuthorizedUser
     { _userId      :: !ClientId
     , _userRefresh :: !RefreshToken
     , _userSecret  :: !Secret
-    } deriving (Eq, Show)
+    } deriving (Eq, Show, Generic)
+
+instance ToJSON AuthorizedUser where
+    toEncoding (AuthorizedUser i r s) =
```

Comment:
By giving `AuthorizedUser` a `Generic` instance and not supplying `toJSON` you will get divergent behaviour between `encode . toJSON` and `encode` which will used `toJSON` and `toEncoding`, respectively. Could you add the `toJSON` implementation, even if it means dropping the `toEncoding`?